code/evaluate/evaluate_model.py

```diff
@@ -94,16 +99,20 @@
 # Paramaterize the matrices on which the models should be compared
 # Add golden data set on which all the model performance can be evaluated
 try:
-    model_list = Model.list(ws)
-    if (len(model_list) > 0):
-        production_model = next(
-            filter(
-                lambda x: x.created_time == max(
-                    model.created_time for model in model_list),
-                model_list,
-            )
-        )
-        production_model_run_id = production_model.run_id
+    firstRegistration = False
+    tag_name = 'experiment_name'
+    try:
+        model_list = get_model_by_tag(
+            model_name, tag_name, exp.name, ws)
+    except Exception as e:
+        firstRegistration = True
```

Comment:
Any exception may occur in `get_model_by_tag` or even in `exp.name`. It does not mean that it's because the list is empty.

Reply:
Simplified the model_helper code a bit, but effectively doing the same thing.  This shouldn't be a problem if `get_model_by_tag` throws an exception in _any_ case - we handle it and only need to care that no model was returned with that tag from the workspace.  If there are other problems it will get picked up in other `try` `except` blocks we have setup.